resources/lib/modules/updates.py

```diff
@@ -405,184 +360,155 @@ def custom_sort_train(self, a, b):
         elif (a_builder > b_builder):
           return +1
 
+    @log.log_function()
     def get_channels(self):
-        try:
-            self.oe.dbg_log('updates::get_channels', 'enter_function', self.oe.LOGDEBUG)
-            channels = []
-            self.oe.dbg_log('updates::get_channels', str(self.update_json), self.oe.LOGDEBUG)
-            if not self.update_json is None:
-                for channel in self.update_json:
-                    channels.append(channel)
-            self.oe.dbg_log('updates::get_channels', 'exit_function', self.oe.LOGDEBUG)
-            return sorted(list(set(channels)), key=cmp_to_key(self.custom_sort_train))
-        except Exception as e:
-            self.oe.dbg_log('updates::get_channels', f'ERROR: ({repr(e)})')
+        channels = []
+        oe.dbg_log('updates::get_channels', str(self.update_json), oe.LOGDEBUG)
+        if not self.update_json is None:
+            for channel in self.update_json:
+                channels.append(channel)
+        return sorted(list(set(channels)), key=cmp_to_key(self.custom_sort_train))
 
+    @log.log_function()
     def do_manual_update(self, listItem=None):
-        try:
-            self.oe.dbg_log('updates::do_manual_update', 'enter_function', self.oe.LOGDEBUG)
+        self.struct['update']['settings']['Build']['value'] = ''
+        update_json = self.build_json(notify_error=True)
+        if update_json is None:
+            return
+        self.update_json = update_json
+        builds = self.get_available_builds()
+        self.struct['update']['settings']['Build']['values'] = builds
+        xbmcDialog = xbmcgui.Dialog()
+        buildSel = xbmcDialog.select(oe._(32020), builds)
+        if buildSel > -1:
+            listItem = builds[buildSel]
+            self.struct['update']['settings']['Build']['value'] = listItem
+            channel = self.struct['update']['settings']['Channel']['value']
+            regex = re.compile(self.update_json[channel]['prettyname_regex'])
+            longname = '-'.join([oe.DISTRIBUTION, oe.ARCHITECTURE, oe.VERSION])
+            if regex.search(longname):
+                version = regex.findall(longname)[0]
+            else:
+                version = oe.VERSION
+            if self.struct['update']['settings']['Build']['value'] != '':
+                self.update_file = self.update_json[self.struct['update']['settings']['Channel']['value']]['url'] + self.get_available_builds(self.struct['update']['settings']['Build']['value'])
+                message = f"{oe._(32188)}: {version}\n{oe._(32187)}: {self.struct['update']['settings']['Build']['value']}\n{oe._(32180)}"
+                answer = xbmcDialog.yesno('LibreELEC Update', message)
```

Comment:
^ hardcoded distro name

Reply:
This is in the existing code
The best would be to move this string (and other hardcoded strings) to strings.po
I leave this to someone else